[DOCS] Improve docs for Windows DOS/UNC paths in `path.*` settings

jrodewig opened the pull request:
Better delineates the paths supported by Windows and Unix-like systems.
Adds more examples for Windows DOS/UNC paths in `path.*` setting snippets.

Closes #57630

Closes #51211

### Preview

Register a snapshot repository: https://elasticsearch_64668.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/snapshots-register-repository.html

Logging: https://elasticsearch_64668.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/logging.html

Path settings: https://elasticsearch_64668.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/important-settings.html#path-settings

jrodewig committed:
[DOCS] Note `path.repo` supports UNC paths in Windows only

elasticmachine commented:
Pinging @elastic/es-distributed (:Distributed/Snapshot/Restore)

elasticmachine commented:
Pinging @elastic/es-docs (>docs)

DaveCTurner commented:
Hmm, tricky, this doesn't quite clear up the confusion in #57630 IMO.

Rather than suggesting that today's examples are canonical can we make it clear that they're only for Unix systems, and for the sake of balance add some Windows-formatted examples too? This applies to both the `path.repo` node setting and the `location` repo setting. We could reasonably do this in  https://www.elastic.co/guide/en/elasticsearch/reference/current/path-settings.html too.

jrodewig committed:
Address feedback

jrodewig committed:
Fixes + rewords

jrodewig commented:
Thanks for the feedback and helping me zoom out on this problem. I've expanded the PR to cover docs that include `path.*`-related snippets and made a few changes:

* Added tab widgets for Unix-like and Windows systems so users can switch based on their platform
* Added more Windows examples, including examples using traditional DOS paths
* Converted several of the YAML arrays to sequences, which I believe are more common

Unfortunately, the tab widgets forced me to spread the content out over several files. I've included some previews that may make it easier to review:

Register a snapshot repository: https://elasticsearch_64668.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/snapshots-register-repository.html

Logging: https://elasticsearch_64668.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/logging.html

Path settings: https://elasticsearch_64668.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/important-settings.html#path-settings

Please let me know if you have any additional feedback.

jrodewig commented:
^ Forgot to tag you @DaveCTurner. Thanks again! 😄

jrodewig committed:
Remove redundant content

jrodewig committed:
Re-add intro

jrodewig committed:
Add note for unix-style paths

jrodewig committed:
Add note for unix-style paths

jrodewig committed:
Add intro

jrodewig committed:
Reword

jrodewig committed:
Reword logging warning

jrodewig committed:
Improve multi data path examples

jrodewig committed:
On second thought, UNC paths in path.data is a bad idea.

jrodewig committed:
Rewords

jrodewig committed:
Reword

jrodewig committed:
Fix typo

jrodewig committed:
Remove unneeded words

jrodewig committed:
Reword

DaveCTurner commented on docs/reference/tab-widgets/register-fs-repo.asciidoc:
```diff
@@ -0,0 +1,109 @@
+    - "F:\\Mount\\Long_term_backups"
+----
+
+For UNC paths, provide the server and share name as a prefix:
```
Suggest combining these two examples, to avoid suggesting that all the paths must be either one or the other.

jrodewig replied:
Combined these with 02e169b7517.

DaveCTurner commented on docs/reference/snapshot-restore/register-repository.asciidoc:
```diff
@@ -100,71 +100,20 @@ are left untouched and in place.
-// TEST[continued]
+include::{es-repo-dir}/tab-widgets/code.asciidoc[]
 
-The following settings are supported:
```
Looks like we lost these settings in the move :smile:

jrodewig replied:
Thanks for pointing this out. I intentionally removed these params as they're also documented in the API reference here: https://www.elastic.co/guide/en/elasticsearch/reference/master/put-snapshot-repo-api.html#put-snapshot-repo-api-request-body

I can re-add if you'd prefer to have those docs in both places, but I figured a link to the API reference would suffice.

DaveCTurner reviewed (changes requested):
That's a great idea, thanks. I left a couple of minor comments but I like the structure here much better now.

jrodewig committed:
Combine DOS/UNC example

jrodewig commented on docs/reference/snapshot-restore/register-repository.asciidoc:
```diff
@@ -100,71 +100,20 @@ are left untouched and in place.
-`max_restore_bytes_per_sec`:: Throttles per node restore rate. Defaults to unlimited. Note that restores are also throttled through <<recovery,recovery settings>>.
-`max_snapshot_bytes_per_sec`:: Throttles per node snapshot rate. Defaults to `40mb` per second.
-`readonly`:: Makes repository read-only.  Defaults to `false`.
-`max_number_of_snapshots`:: Limits the maximum number of snapshots that the repository may contain. Defaults to `500`. Note that snapshot repositories do not
-scale indefinitely in size and might lead to master node performance and stability issues if they grow past a certain size. We do not recommend increasing this setting.
-Instead you should delete older snapshots or use multiple repositories.
```
This param is not documented in the API reference docs as we don't actually recommend setting it.

DaveCTurner replied:
We should really document it anyway, although I don't think it's been backported to 7.x yet.

jrodewig committed:
Make path plural

jrodewig commented:
Thanks again for the feedback @DaveCTurner. I combined the DOS/UNC examples.

I intentionally removed several of the parameters for the put snapshot repo API as they're also documented here:
https://www.elastic.co/guide/en/elasticsearch/reference/master/put-snapshot-repo-api.html#put-snapshot-repo-api-request-body

Let me know if you feel those should be re-added.

DaveCTurner reviewed (approved):
LGTM

jrodewig commented:
Thanks @DaveCTurner. I'll open a separate PR to address the issues in the put snapshot repository API reference docs.
